create-react-app client... c'est ok ?

aaadryyy opened the pull request:
J'ai fait ça dans le doute, n'ayant jamais branché React à un projet existant au préalable.

aaadryyy committed:
create-react-app client

adipasquale commented on .idea/encodings.xml:
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
you should add the whole .idea folder to .gitignore so that it's not commited, as it's specific to your IDE

adipasquale commented on api/package-lock.json:
```diff
@@ -0,0 +1,96 @@
+{
```
this file should not be in the `api/` folder. Actually, maybe it should not be commited at all, I think you have to decide between NPM or Yarn. Yarn uses `yarn.lock`, NPM uses `package-lock.json`. There should be only one of the two.

aaadryyy replied:
That's weird, I created the project with npm only.

adipasquale commented on client/.gitignore:
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
I'm guessing this is the default `.gitignore` file generated upon creating the react app. Because we are here in a multi-project repository, you'll have to merge the contents of this new file into the existing `.gitignore` that's at the root level. Also, you'll have to prefix most of the paths with `client/` because the paths will be relative from the root.

adipasquale commented on client/.gitignore:
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
```
as you can see, node_modules are ignored by default

adipasquale commented on client/package.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "client",
```
would be nice to provide a more explicit name like `insee_brut_client`

adipasquale commented on client/public/index.html:
```diff
@@ -0,0 +1,40 @@
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Let's change that title

aaadryyy replied:
Not sure what you mean by "brut" but "INSEE Brut" will make the deal for now :)

adipasquale commented on client/public/manifest.json:
```diff
@@ -0,0 +1,15 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Same here, please customise these fields

aaadryyy replied:
"short_name": "INSEE Brut"
"name": "Mirror site for INSEE data aimed to be exhaustive, faster and easier to browse."

adipasquale commented on package-lock.json:
```diff
@@ -0,0 +1,3 @@
+{
```
I don't think you should be commiting any new files on the root level. This `package-lock.json` appears like an error.

adipasquale commented on package.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "insee-brut",
+  "version": "1.0.0",
+  "description": "Mirror site for INSEE data aimed to be exhaustive, faster and easier to browse.",
```
ah ok You customised this root package.json file, but I think it should not be here but inside the `client` folder. I'm pretty sure there should not be two different `package.json` files, one at the root level and one in the `client` folder

adipasquale reviewed (commented):
I made a few comments in the source code but in short : 
- make sure to fill in the root `/.gitignore` with the paths that should be ignore, most importantly with `/client/node_modules`.
- for the dependency management, make sure that you're only committing one `/client/yarn.lock` file instead of 4 in total, with root levels `/package-lock.json`, `/yarn.lock`, and a `/client/package-lock.json`. I'm guessing you ran both `npm install` and `yarn install` locally, you have to pick one :) let's go with yarn !

aaadryyy committed:
FIX

aaadryyy committed:
+ Tachyons

adipasquale commented:
@aaadryyy it's not yet completely ok sorry :
- you didn't add .idea to the root `.gitignore` file. 
- you didn't move the content of the `client/.gitignore` file into the root one, but just purely deleted it. We need at the very least the `client/node_modules` line in it.
- when you add paths to the `.gitignore` file, it doesn't remove them from the repository. You have to use `git rm --cached` to do it. The option makes sure that the files are not deleted locally, just uncommited from the repository.

aaadryyy commented:
ok thanks for your patience :

- I just added `.idea` to the root `.gitignore`, along with the previous  `client/.idea` add.
- In my previous commit, I added all the content of the `client/.gitignore` to the root `.gitignore` ( `client/node_modules` included), that's why I deleted it.
- `git rm --cached filename` did not work for a majority of files. Did `git rm --cached . -r` but deleted plenty of file. Guess it was not the right thing to do...

aaadryyy committed:
FIX .gitignore + cached file

aaadryyy commented:
ok thanks for your patience :

- I just added `.idea` to the root `.gitignore`, along with the previous `client/.idea` add.

- In my previous commit, I added all the content of the `client/.gitignore` to the root `.gitignore` ( `client/node_modules` included), that's why I deleted it.

- git `rm --cached filename` did not work for the majority of files. Did `git rm --cached . -r` but deleted plenty of file. Guess it was not the right thing to do...

adipasquale commented:
hey @aaadryyy merry christmas 🎅 

I really don't mean to be annoying but it's not yet ready. have you looked at the 'files changed' tab ? https://github.com/adipasquale/insee-brut/pull/9/files 
You can see that there are far too many files being committed in your PR, amongst which the `node_modules` folder

It can be a bit annoying to uncommit files. Since you have almost only ran `create-react-app` so far, maybe a good solution would be to re-start from scratch. delete this branch and PR, go back to master, branch and re-do `create-react-app`. and make sure to : 
- update the root `.gitignore` before anything.
- be very careful when you're git adding files. I personnally never use `git add .` as it can result in files committed by error, like here. I try and add new files manually one by one. In this specific PR case, since you're adding quite a few files you may have ot make some exceptions to this rule, but you can maybe limit yourself to `git adding` only sub-directories or something like that. 
- That's for new files, and for files updates, I usually use `git add -p`. This will take you through all the updates you made chunk by chunk (like 10 lines by 10 lines), and you can decide if you want to keep each change or not. In this PR you won't have many updated files, since you're mostly adding new ones.
- **ALWAYS run `git diff --cached` before `git commit`**. It will show you the exhaustive summary of changes you're about to commit (files added, removed, updated).

aaadryyy commented:
🎄🎄 🎄 🎅 🎄🎄🎄

Haha you're not annoying at all, I am learning a lot!
But if you get annoyed please tell me 😉 
Did a new PR following your indications

adipasquale commented:
ok ! I'm closing this one
